core/src/main/scala/com/microsoft/azure/synapse/ml/causal/ComputeResidualTransformer.scala

```diff
@@ -0,0 +1,92 @@
+package com.microsoft.azure.synapse.ml.causal
+
+import com.microsoft.azure.synapse.ml.codegen.Wrappable
+import com.microsoft.azure.synapse.ml.core.schema.SchemaConstants
+import com.microsoft.azure.synapse.ml.logging.BasicLogging
+import org.apache.spark.ml.Transformer
+import org.apache.spark.ml.linalg.Vector
+import org.apache.spark.ml.param._
+import org.apache.spark.ml.util.{DefaultParamsReadable, DefaultParamsWritable, Identifiable}
+import org.apache.spark.sql.functions.{col, udf}
+import org.apache.spark.sql.{DataFrame, Dataset}
+import org.apache.spark.sql.types._
+
+class ComputeResidualTransformer(override val uid: String) extends Transformer
+  with DefaultParamsWritable
+  with Wrappable
+  with BasicLogging {
```

Comment:
There is a `HasOutputCol` trait from `org.apache.spark.ml.param.shared` package. Use that instead of creating your own outputCol param.
```suggestion
class ComputeResidualTransformer(override val uid: String) extends Transformer
  with DefaultParamsWritable
  with Wrappable
  with HasOutputCol
  with BasicLogging
```